test/unit/lib/sqs.spec.js

```diff
@@ -134,20 +151,176 @@ describe('SQS Utilities', function() {
+          expect(sqsMock.sendMessage.args[0][0]).to.deep.equal({
+            MessageBody: compress(JSON.stringify(payload)).toString('base64'),
+            QueueUrl: queueUrl,
+            MessageAttributes: { foo: 'bar' }
+          });
+        });
+    });
+
+    it('should send sqs-only string messages when payload size is <256kb', function() {
+      const payload = 'look-ma-its-a-string';
+      return sqsInstance.extendedSend({ queueName: 'queue', s3Bucket: 'test', payload, attrs: { foo: 'bar' } })
+        .then((res) => {
+          expect(res).to.deep.equal({ res: result, extended: false });
+          expect(sqsMock.sendMessage.callCount).to.equal(1);
+          expect(sqsMock.sendMessage.args[0][0]).to.deep.equal({
+            MessageBody: compress(payload).toString('base64'),
+            QueueUrl: queueUrl,
+            MessageAttributes: { foo: 'bar' }
+          });
+        });
+    });
+
+    it('should send s3-extended messages when payload size is >=256kb', function() {
+      let payload;
+
+      for (let i = 0; i < 300000; i++) {
+        payload += Math.random().toString(36).substr(2, 1);
+      }
+
+      sinon.stub(Math, 'random');
+      Math.random.returns(.5);
+
+      return sqsInstance.extendedSend({ queueName: 'queue', s3Bucket: 'test', payload })
+        .then(() => {
+          expect(s3Mock.upload).to.have.been.calledWithMatch({
+            Bucket: 'test'
+          });
+          expect(s3Mock.upload.args[0][0].Key).to.match(/25\/[0-9A-F]{8}-[0-9A-F]{4}-4[0-9A-F]{3}-[89AB][0-9A-F]{3}-[0-9A-F]{12}\.json\.gz/i);
+        });
+    });
+  });
+
+  describe('extendedRetrieve', function() {
+    beforeEach(function() {
+      sqsMock.receiveMessageAsync = sinon.stub().resolves({ Messages: [
+        {
+          Body: compress(JSON.stringify({ expect: 'a passing test!' }))
+        }
+      ]});
+    });
+
+    it('should handle empty receives', function() {
+      sqsMock.receiveMessageAsync = sinon.stub().resolves({ Messages: [] });
+
+      return sqsInstance.extendedRetrieve({ queueName: 'queue' })
+        .then((res) => {
+          expect(res.length).to.equal(0);
+          expect(sqsMock.receiveMessageAsync.callCount).to.equal(1);
+        });
+    });
+
+    it('should retrieve messages without s3', function() {
+      return sqsInstance.extendedRetrieve({ queueName: 'queue' })
+        .then((res) => {
+          expect(res[0].body).to.deep.equal({ expect: 'a passing test!' });
+          expect(sqsMock.receiveMessageAsync.callCount).to.equal(1);
+          expect(sqsMock.receiveMessageAsync.args[0][0]).to.deep.equal({
+            MaxNumberOfMessages: 10,
+            QueueUrl: queueUrl,
+            WaitTimeSeconds: 20,
+            MessageAttributeNames: [
+              'EXTENDED_S3_BUCKET',
+              'EXTENDED_S3_KEY'
+            ],
+            VisibilityTimeout: 300 // default timeout
+          });
+        });
+    });
+
+    it('should retrieve messages with s3', function() {
```

Comment:
is this test even correct? I would've thought it is returning  "true" in message body.  Also you probably want to assert that the S3.getObject is called

Reply:
it's correct because it resolved the s3 message contents (something truthy has to be set to the sqs body).